conversions/octal_to_hexadecimal.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * @brief Octal to hexadecimal conversion by scanning user input
+ * @details
+ * The octalToHexadecimal function take the octal number as long
+ * return a string hexadecimal value after conversion
+ * @author [Rachit Bhalla](https://github.com/rachitbhalla)
+ */
+#include <assert.h> // for assert
+#include <math.h> // for pow function
+#include <stdio.h> // for scanf and printf functions
+#include <stdlib.h> // for malloc and free functions
+#include <string.h> // for strcmp function
+
+/**
+ * @brief Convert octal number to decimal number
+ * @param octalValue is the octal number that needs to be converted
+ * @returns a decimal number after conversion
+ */
+long octalToDecimal(long octalValue){
+    long decimalValue = 0;
+    int i = 0;
+    while (octalValue) {
+        // Extracts right-most digit, multiplies it with 8^i, and increment i by 1
+        decimalValue += (long)(octalValue % 10) * pow(8, i++);
+        // Shift right in base 10
+        octalValue /= 10;
+    }
+    return decimalValue;
+}
+
+/**
+ * @brief Convert octal number to hexadecimal number
+ * @param octalValue is the octal number that needs to be converted
+ * @returns a hexadecimal value as a string after conversion
+ */
+char *octalToHexadecimal(long octalValue){
+    char *hexadecimalValue = malloc(256 * sizeof(char));
```

Comment:
Please mention in the function description that the dynamically allocated memory needs to be freeed by the calling function.

Reply:
Updated! Pls check.

Reply:
A simpler approach would be to avoid `malloc` altogether.
```c
#define MAX_HEX_LENGTH 256;
static char hexadecimalValue[MAX_HEX_LENGTH];
memset(hexadecimalValue, '\0\, MAX_HEX_LENGTH);
```
Since the memory is not dynamically allocated, there is no need to call `free`. To ensure that the memory is retained outside the heap, the variable is declared as static.